File: tictactoe/src/main/resources/tictactoe/solutions/MyTicTacToePlayer.java

import tictactoe.wrappers.java.TicTacToePlayer;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

/**
 * TicTacToe player in Java
 *
 * @author José Carlos Paiva <code>josepaiva94@gmail.com</code>
 */
public class MyTicTacToePlayer extends TicTacToePlayer {

    Random random = null;

    @Override
    public String getName() {
        return "José Carlos Paiva";
    }

    @Override
    public void init() {
        random = new Random();
    }

    @Override
    public void execute() {
        play(generateMove());
    }

    /**
     * Generate a position to play the piece
     *
     * @return position to play the piece
     */
    int generateMove() {
        List<Integer> possibleMoves = new ArrayList<Integer>();

        for (int i = 1; i <= 9; i++)
            if (isFree(i))
                possibleMoves.add(i);

        if (possibleMoves.isEmpty())
            return 0;

        return possibleMoves.get(random.nextInt(possibleMoves.size()));
    }
}

File: tictactoe/src/main/resources/tictactoe/solutions/MyTicTacToePlayer.js

// ------------------------ Skeleton --------------------------------

load('tictactoe/wrappers/js/TicTacToePlayer.js');

(function (exports) {'use strict';

    /**
     * TicTacToe player in Javascript
     *
     * @author José Carlos Paiva <code>josepaiva94@gmail.com</code>
     */
    function MyTicTacToePlayer() {
        TicTacToePlayer.call(this);
    }

    MyTicTacToePlayer.prototype = Object.create(TicTacToePlayer.prototype);

    // ---------------------- Your code below ---------------------------

    /**
     * Get the name of the player
     *
     * @returns {string} name of the player
     */
    MyTicTacToePlayer.prototype.getName = function () {
        return 'Professor X';
    };

    /**
     * Initialize the player
     */
    MyTicTacToePlayer.prototype.init = function () {
    };

    /**
     * Execute player action
     */
    MyTicTacToePlayer.prototype.execute = function () {
        this.play(this.generateMove());
    };

    /**
     * Generates randomly a valid move
     *
     * @returns {number} a valid move
     */
    MyTicTacToePlayer.prototype.generateMove = function () {

        var possibleMoves = [];

        var i = 1;
        while (i <= 9) {
            if (this.isFree(i))
                possibleMoves.push(i);
            i++;
        }

        if (possibleMoves.length === 0)
            return 0;

        return possibleMoves[getRandomInt(possibleMoves.length)];
    };

    /**
     * Get a random int between 0 and max
     *
     * @param max {number} upper limit
     * @returns {number} random int between 0 and max
     */
    function getRandomInt(max) {
        return Math.floor(Math.random() * Math.floor(max));
    }

    // ------------------------ Skeleton --------------------------------

    exports.Player = MyTicTacToePlayer;
}(this));

